x/concentrated-liquidity/incentives_test.go

```diff
@@ -358,6 +368,30 @@ func (s *KeeperTestSuite) TestCreateAndGetUptimeAccumulatorValues() {
 }
 
 func (s *KeeperTestSuite) TestCalcAccruedIncentivesForAccum() {
+	incentiveRecordOneWithDifferentStartTime := createIncentiveRecord(
+		incentiveRecordOne.IncentiveDenom,
+		incentiveRecordOne.RemainingAmount,
+		incentiveRecordOne.EmissionRate,
+		incentiveRecordOne.StartTime.Add(10),
+		incentiveRecordOne.MinUptime,
+	)
+
+	incentiveRecordOneWithDifferentMinUpTime := createIncentiveRecord(
+		incentiveRecordOne.IncentiveDenom,
+		incentiveRecordOne.RemainingAmount,
+		incentiveRecordOne.EmissionRate,
+		incentiveRecordOne.StartTime,
+		testUptimeTwo,
+	)
+
+	incentiveRecordOneWithDifferentDenom := createIncentiveRecord(
+		testDenomTwo,
+		incentiveRecordOne.RemainingAmount,
+		incentiveRecordOne.EmissionRate,
+		incentiveRecordOne.StartTime,
+		incentiveRecordOne.MinUptime,
+	)
```

Comment:
nit: to reduce the amount of boilerplate, we can use a decorator pattern:

```go
func withDenom(record types.IncentiveRecord, denom string) types.IncentiveRecord {
	record.IncentiveRecord = denom
}
```

then, we can call this like:
```go
incentiveRecordOneWithDifferentDenom := withDenom(incentiveRecordOne, testDenomTwo)
```

Reply:
exactly what i was looking for